lte/gateway/c/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -0,0 +1,323 @@
+
+  if (found_ue_id == ue_context_map.end()) {
+    return NULL;
+  } else {
+    return found_ue_id->second;
+  }
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_insert_smf_context()                                      **
+ **                                                                        **
+ ** Description: Insert smf context in the vector                          **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+smf_context_t* amf_insert_smf_context(
+    ue_m5gmm_context_s* ue_context, uint8_t pdu_session_id) {
+  smf_context_t smf_context;
+  std::vector<smf_context_t>::iterator i;
+  int j = 0;
+
+  for (i = ue_context->amf_context.smf_ctxt_vector.begin();
+       i != ue_context->amf_context.smf_ctxt_vector.end(); i++, j++) {
+    OAILOG_INFO(LOG_AMF_APP, "insert smf_ctx j%d", j);
+    if (i->smf_proc_data.pdu_session_identity.pdu_session_id ==
+        pdu_session_id) {
+      ue_context->amf_context.smf_ctxt_vector.at(j) = smf_context;
+      return ue_context->amf_context.smf_ctxt_vector.data() + j;
+    }
+  }
+
+  // add new element to the vector
+  OAILOG_INFO(
+      LOG_AMF_APP, "insert smf_ctx_vector_data%d",
+      ue_context->amf_context.smf_ctxt_vector.data());
+  OAILOG_INFO(
+      LOG_AMF_APP, "insert smf_ctx_vector_size%d",
+      ue_context->amf_context.smf_ctxt_vector.size());
+  ue_context->amf_context.smf_ctxt_vector.push_back(smf_context);
+
+  OAILOG_INFO(
+      LOG_AMF_APP, "insert smf_ctx_vector_data%d",
+      ue_context->amf_context.smf_ctxt_vector.data());
+  return ue_context->amf_context.smf_ctxt_vector.data() + j;
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_smf_context_exists_pdu_session_id()                       **
+ **                                                                        **
+ ** Description: Get the smf context from the vector                       **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+smf_context_t* amf_smf_context_exists_pdu_session_id(
+    ue_m5gmm_context_s* ue_context, uint8_t id) {
+  std::vector<smf_context_t>::iterator i;
+  int j = 0;
+  for (i = ue_context->amf_context.smf_ctxt_vector.begin();
+       i != ue_context->amf_context.smf_ctxt_vector.end(); i++, j++) {
+    if (i->smf_proc_data.pdu_session_identity.pdu_session_id == id) {
+      return ue_context->amf_context.smf_ctxt_vector.data() + j;

```

Comment:
what is the purpose of using `j`? You can directly use the iterator to retrieve the element.